internal/cloud/realm/secrets_ext_test.go

```diff
@@ -0,0 +1,60 @@
+package realm_test
+
+import (
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	u "github.com/10gen/realm-cli/internal/utils/test"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestSecrets(t *testing.T) {
+	u.SkipUnlessRealmServerRunning(t)
+
+	t.Run("should fail without an auth client", func(t *testing.T) {
+		client := realm.NewClient(u.RealmServerURL() + "")
+
+		_, err := client.Secrets(primitive.NewObjectID().Hex(), primitive.NewObjectID().Hex())
+		assert.Equal(t, realm.ErrInvalidSession{}, err)
+	})
+
+	t.Run("with an active session ", func(t *testing.T) {
+		client := newAuthClient(t)
+		groupID := u.CloudGroupID()
+
+		secretName := "secretName"
+		secretValue := "secretValue"
+
+		testApp, appErr := client.CreateApp(groupID, "secrets-test", realm.AppMeta{})
+		assert.Nil(t, appErr)
+		defer client.DeleteApp(groupID, testApp.ID)
+
+		assert.Nil(t, client.Import(groupID, testApp.ID, map[string]interface{}{
+			app.NameAuthProviders: []map[string]interface{}{
+				{"name": "api-key", "type": "api-key"},
+				{"name": "local-userpass", "type": "local-userpass", "config": map[string]interface{}{
+					"resetPasswordUrl":     "http://localhost:8080/reset_password",
+					"emailConfirmationUrl": "http://localhost:8080/confirm_email",
+				}},
+			},
+		}))
+
+		t.Run("should have no secrets upon app initialization", func(t *testing.T) {
+			secrets, secretsErr := client.Secrets(groupID, testApp.ID)
+			assert.Nil(t, secretsErr)
+			assert.Equal(t, 0, len(secrets))
+		})
+		t.Run("should create a secret", func(t *testing.T) {
+			assert.Nil(t, client.CreateSecret(groupID, testApp.ID, secretName, secretValue))
+		})
+		t.Run("should list a created secret", func(t *testing.T) {
+			secrets, secretsErr := client.Secrets(groupID, testApp.ID)
+			assert.Nil(t, secretsErr)
+			assert.Equal(t, 1, len(secrets))
```

Comment:
these two asserts can go away with just:

```go
assert.Equal(t, []Secret{{secretName, secretValue}}, secrets)
```

Reply:
The `client.Secrets()` call returns an object with an ID and Name so I went with the 2-liner here since we don't check the id. We could modify `client.CreateSecret` to return an ID, capture the ID in an earlier test and then check the assert here. I felt CreateSecret was fine just returning an error if we don't display the ID of the created secret in the command output

Reply:
sorry, I meant turn
```go
assert.Equal(t, 1, len(secrets))
assert.Equal(t, secretName, secrets[0].Name)
```
into
```go
assert.Equal(t, []Secret{{secretName, secretValue}}, secrets)
```

Reply:
I may have misinterpreted but I was referring to those lines as well. the realm.Secret object has `ID` and 'Name` as opposed to `Name` and `Value` so there would be an assert error on `assert.Equal(t, []Secret{{secretName, secretValue}}, secrets)` since `secretValue` won't match the ID field returned bt the `Secrets()` call

Reply:
I gotcha now...sorry should have read closer (I was getting request/response mixed up therE).  this is fine as is then!

Reply:
No worries, all good!

Reply:
ok last bit on this: update `CreateSecret` to return `(Secret, error)` (since that's actually what the API returns to us).  from there, you can assert on `.Name` being correct in the "create" test, and you can assert on a whole `[]Secret` slice in the ensuing "find all created" test